client/state/sites/vouchers/actions.js

```diff
@@ -75,6 +79,68 @@ export const vouchersRequestFailureAction = ( siteId, error ) => {
 	};
 };
 
+export const vouchersAssignReceiveAction = ( siteId, serviceType, voucher ) => {
+	return {
+		type: SITE_VOUCHERS_ASSIGN_RECEIVE,
+		siteId,
+		serviceType,
+		voucher
+	};
+};
+
+/**
+ * Action creator function
+ *
+ * Return SITE_VOUCHERS_ASSIGN_REQUEST action object
+ *
+ * @param {Number} siteId - side identifier
+ * @param {String} serviceType - service type
+ * @return {Object} siteId - action object
+ */
+export const vouchersAssignRequestAction = ( siteId, serviceType ) => {
+	return {
+		type: SITE_VOUCHERS_ASSIGN_REQUEST,
+		siteId,
+		serviceType
+	};
+};
+
+/**
+ * Action creator function
+ *
+ * Return SITE_VOUCHERS_ASSIGN_REQUEST_SUCCESS action object
+ *
+ * @param {Number} siteId - side identifier
+ * @param {String} serviceType - service type
+ * @return {Object} siteId - action object
+ */
+export const vouchersAssignRequestSuccessAction = ( siteId, serviceType ) => {
+	return {
+		type: SITE_VOUCHERS_ASSIGN_REQUEST_SUCCESS,
+		siteId,
+		serviceType
+	};
+};
+
+/**
+ * Action creator function
+ *
+ * Return SITE_VOUCHERS_ASSIGN_REQUEST_FAILURE action object
+ *
+ * @param {Number} siteId - site identifier
+ * @param {String} serviceType - service type
+ * @param {Object} error - error message according to REST-API error response
+ * @return {Object} action object
+ */
+export const vouchersAssignRequestFailureAction = ( siteId, serviceType, error ) => {
```

Comment:
Just curious, but is there a reason we are prefixing all of the function names with `vouchers` here? these are all hidden under the "module namespace" if you will, so there's no need to copy the faux-namespacing that we use in old versions of PHP inside the JavaScript world - it's just more redundant typing that tends to obscure through verboseness rather than help.